projects/example-app/src/app/reducers/index.ts

```diff
@@ -69,7 +69,7 @@ export const metaReducers: MetaReducer<State>[] = !environment.production
  * Layout Reducers
  */
 export const selectLayoutState = createFeatureSelector<State, fromLayout.State>(
```

Comment:
just one generic is fine `createFeatureSelector<fromLayout.State>(...)`
Typically there is no global `State` and many features are lazy-loaded, so there cannot be a single interface that connects them all.

Reply:
If this is to be added I guess it would imply changes to all other places where FeatureSelectors are being created e.g. https://github.com/ngrx/platform/blob/4b48e51f95/projects/example-app/src/app/auth/reducers/index.ts#L29 ??

Reply:
I think so, yes. Maybe in the follow-up PRs.
Also, after PR https://github.com/ngrx/platform/pull/2325 lands there would be some more changes
@timdeschryver @brandonroberts hope you don't mind.

Reply:
The 2 generics are there to provide type safety, but you have to extend the Feature state with the Root State:

```ts
export interface State extends fromRoot.State {
  [booksFeatureKey]: BooksState;
}
```

But this was before the usage of feature keys.

I think we could indeed refactor this, but I don't got a strong opinion on this. 
Sometimes it will be needed to extend from the root, if you're combining selectors from the feature with the root.

We can take this one in a later PR and merge this PR if everything is OK with it.

Reply:
> Sometimes it will still be needed to extend from the root, this is when you're combining selectors from the feature with the root.

There are workarounds for this :P